Fix #2774: Show export success modal only after all files are exported

tomasmbrito opened the pull request:
Fix #2774: Show export success modal only after all files are exported

Previously, clicking "Run Export" would immediately show the success message, even before the export process had finished.

Now, the success modal only appears after all files have actually been exported.

I also implemented a unit test to validate this behavior, but for the test to run, it is necessary to add the dependency:
"@vue/test-utils": "^1.3.0" to the apps/studio/package.json file.

I wasn’t sure if I could change the package.json, so I opted not to include it in the commit.

tomasmbrito commented:
Hi, just wanted to let you know that for the unit test I added to work properly, the following dev dependency needs to be included in apps/studio/package.json:
"@vue/test-utils": "^1.3.0"
I didn’t include it in the commit since I wasn’t sure if it was okay to modify that file. But I can add it if needed.

rathboma commented:
@tomasmbrito Yes, please include the dependency if we need it for the test to work!

rathboma commented:
@not-night-but added you as a reviewer!

tomasmbrito commented:
I’ve added the dependency to package.json as asked.
The yarn.lock file was also updated automatically when installing it but it’s not directly related to the bug fix or the new test.

not-night-but reviewed (approved):
Looks good to me! Thanks for doing this.
